### PR TITLE
[BUGFIX] Correct handling of language overlays

### PR DESCRIPTION
**Pull request by NamelessCoder**
This patch removes nearly all VHS content pulling logic, replacing it with the frontend-only ContentObjectRenderer method to getRecords().

The fix affects:

* Content overlay modes `content_fallback` and `strict` now correctly respected
* Workspace and preview overlay is now done by core internals
* Language menu generation uses same API and gets overlayed for language/workspace preview
* Content sliding slideCollect no longer overrides slide depth if slide depth is set
* Content sliding behavior reversal functionality ("slideCollectReverse" argument) restored

Close: #1297
Close: #1321
Close: #1323
Close: #761
Close: #1049

### CODE REVIEW

**Comment by RosenbauerSolbach** [comment]
I tried to update vhs 4.2 to 4.3.1 today. TYPO3 version is 8.7.6. 

<v:content.render column="0"/> 
This renders every column, not only colPos=0 in new version 4.3.1

If I set any value, except 0 it work as it should
<v:content.render column="2" />

Maybe that 0 acts like a FALSE?

**Comment by NamelessCoder** [comment]
@RosenbauerSolbach Fixed in dev-development, not yet released to TER, has several other open issues and is unrelated to this change. Please check these things a bit better before commenting on pull requests - this isn't your fault specifically but over the last week or so I've had probably 20 people asking and reporting this very same issue and honestly, it gets old very quickly.

**Comment by NamelessCoder** [comment]
@cedricziel @bjo3rnf I need your help guys ;) if you can verify this solution I will pay you back with a VHS 4.3.2 release right after!

**Comment by tantegerda1** [comment]
Not that I can "verify", but to give additional feedback: I have VHS with this pull request applied running on a live website, and indeed it 
- did solve my problem (workspace preview)
- does work properly, especially regarding content retrieval and languages
So, not a formal "verify" but a "works well for me" +1 :D

**Comment by NamelessCoder** [comment]
Thanks @tantegerda1 - that's about as "verified" as it gets and if you switch to the "files" tab you should also be able to throw -1/+1 as a quicker way to give feedback ;)

File: Classes/ViewHelpers/Content/AbstractContentViewHelper.php [hunk 1]
```diff
@@ -102,8 +102,8 @@ protected function getContentRecords()
 
         $pageUid = $this->getPageUid();
 
+        $contentRecords = $this->getSlideRecordsFromPage($pageUid, $limit);
         if ((integer) $this->arguments['slide'] === 0) {
-            $contentRecords = $this->getSlideRecordsFromPage($pageUid, $limit);
```
Comment (bjo3rnf) [inline Classes/ViewHelpers/Content/AbstractContentViewHelper.php]: Empty branch here?

Comment (NamelessCoder) [reply Classes/ViewHelpers/Content/AbstractContentViewHelper.php]: Fixed :)

**Comment by NamelessCoder** [comment]
@bjo3rnf if that's the only note you had then I think we're ready to go - please merge if you agree. I'll then prepare 4.3.2 for release.

**Comment by dr-bock** [comment]
After this fix, content.render ViewHelper is buggy in our TYPO3-Installation (8.7.4).

{v:content.render(contentUids: {0: myContentElementUid})}

does not render anything. It creates an sql with "where pid in (-1)". e.g.

SELECT * FROM `tt_content` WHERE (`tt_content`.`uid` IN (164631)) AND (`tt_content`.`pid` IN (-1)) AND ((`tt_content`.`sys_language_uid` IN (0, -1)) OR ((`l18n_parent` = 0) AND (`tt_content`.`sys_language_uid` = 0))) AND ((`tt_content`.`deleted` = 0) AND (`tt_content`.`t3ver_state` <= 0) AND (`tt_content`.`hidden` = 0) AND (`tt_content`.`starttime` <= 1506000300) AND ((`tt_content`.`endtime` = 0) OR (`tt_content`.`endtime` > 1506000300)) AND (((`tt_content`.`fe_group` = '') OR (`tt_content`.`fe_group` IS NULL) OR (`tt_content`.`fe_group` = '0') OR (FIND_IN_SET('0', `tt_content`.`fe_group`)) OR (FIND_IN_SET('-1', `tt_content`.`fe_group`)))))

**Comment by NamelessCoder** [comment]
@dr-bock 

> SELECT * FROM tt_content WHERE (tt_content.uid IN (164631))

Appears it would select your exact record. The other conditions that get applied all come from TYPO3 itself and include things like enableFields, user permissions and workspace preview conditions. Perhaps one or more of those conditions are preventing the record from showing up?

**Comment by dr-bock** [comment]
Yes, the ViewHelper should render the exact record. And until this Bugfix, it worked correct.
Call it with {v:content.render(contentUids: {0: 164631})}
The above sql-query is the exact query, ContentObjectRenderer build by calling the ViewHelper. And "where pid in (-1)" could never work. All other where-conditions are right.

In AbstractContentViewHelper you call cObj->getRecords with "pidInList => -1"
Don't know if this is the bug or if its in the core. I have not debugged the entire core functions today.

**Comment by NamelessCoder** [comment]
Fixed in dev-development, you may want to apply that patch if your project depends on pulling content by UID!

**Pull request merged**